### PR TITLE
Some v26 features

### PR DESCRIPTION
**Pull request by sisuresh**
### What

Adapt to CAP-0078, CAP-0079 and CAP-0080

### CODE REVIEW

File: soroban-sdk/src/tests/crypto_bn254.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        bn254.fr_pow(&U256::from_u32(&env, 5).into(), 2),
+        U256::from_u32(&env, 25).into()
+    );
+
+    // inv: inv(13) * 13 == 1
+    let inverse_13 = bn254.fr_inv(&U256::from_u32(&env, 13).into());
+    assert_eq!(
+        bn254.fr_mul(&inverse_13, &U256::from_u32(&env, 13).into()),
+        U256::from_u32(&env, 1).into()
+    );
```
Comment (jayz22) [inline soroban-sdk/src/tests/crypto_bn254.rs:142]: let's also add a test case for fr_inv with zero input.

File: soroban-sdk/src/tests/crypto_bn254.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    // 0*G + 0*G = 0
+    let vp: Vec<Bn254G1Affine> = vec![&env, one.clone(), one.clone()];
+    let vs: Vec<Fr> = vec![
+        &env,
+        U256::from_u32(&env, 0).into(),
+        U256::from_u32(&env, 0).into(),
+    ];
+    let res = bn254.g1_msm(vp, vs);
+    assert_eq!(res, zero);
```
Comment (jayz22) [inline soroban-sdk/src/tests/crypto_bn254.rs:63]: Can we also add g1_msm error case, i.e. mismatched vector lengths and empty vectors.

File: soroban-sdk/src/tests.rs [hunk 3]
```diff
@@ -34,6 +34,7 @@ mod contractimport;
 mod contractimport_with_error;
 mod cost_estimate;
 mod crypto_bls12_381;
+mod crypto_bn254;
```
Comment (jayz22) [inline soroban-sdk/src/tests.rs:37]: This branch is out-of-sync with main, on which there are more bn254 Fp/Fr arithmetic tests. We should move those tests to the new file.

File: soroban-sdk/src/tests/crypto_bn254.rs (hunk 1 above)
Comment (sisuresh) [reply soroban-sdk/src/tests/crypto_bn254.rs:142]: Done

File: soroban-sdk/src/tests/crypto_bn254.rs (hunk 2 above)
Comment (sisuresh) [reply soroban-sdk/src/tests/crypto_bn254.rs:63]: Done

File: soroban-sdk/src/tests.rs (hunk 3 above)
Comment (sisuresh) [reply soroban-sdk/src/tests.rs:37]: We should do this as a follow up when v26-preview is updated.
